graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/fetchers/admin/DeploySchemaFetcherBase.java

```diff
@@ -101,6 +97,23 @@ protected DeploySchemaResponseDto get(
     return response;
   }
 
+  private void checkPermissions(StargateGraphqlContext context, String keyspaceName)
+      throws UnauthorizedException {
+
+    AuthorizationService authorizationService = context.getAuthorizationService();
+    AuthenticationSubject subject = context.getSubject();
+    // Ensure we can read the data model to validate the schema:
+    authorizationService.authorizeSchemaRead(
```

Comment:
is this for preventing a later error if the user has a write, but doesn't have read permission?

Reply:
Is it really possible that somebody could have write, but not read permissions?

Reply:
My bad, I need to revisit this, there are a number of issues:
* the check at line 113 should be an `authorizeDataWrite` instead of `authorizeSchemaWrite`. It's only about writing the new row in `stargate_graphql.schema_source`.
* this check is about reading the user keyspace's model (which tables are present, what their columns are, etc). But in fact it should also check for write privilege, since the `deploySchema` operation can modify the model (create tables, add columns, etc).
    I'm not sure if write access implies read access.